crates/bench/src/spacetime_raw.rs

```diff
@@ -0,0 +1,158 @@
+            match table_style {
+                TableStyle::Unique => {
+                    self.db
+                        .create_index(tx, IndexDef::new("id".to_string(), table_id, 0, true))?;
+                }
+                TableStyle::NonUnique => (),
+                TableStyle::MultiIndex => {
+                    for (i, column) in T::product_type().elements.iter().enumerate() {
+                        self.db.create_index(
+                            tx,
+                            IndexDef::new(column.name.clone().unwrap(), table_id, i as u32, false),
+                        )?;
+                    }
+                }
+            }
+
+            Ok(table_id)
+        })
+    }
+
+    fn clear_table(&mut self, table_id: &Self::TableId) -> ResultBench<()> {
+        self.db.with_auto_commit(|tx| {
+            self.db.clear_table(tx, *table_id)?;
+            Ok(())
+        })
+    }
+
+    fn count_table(&mut self, table_id: &Self::TableId) -> ResultBench<u32> {
+        self.db
+            .with_auto_commit(|tx| Ok(self.db.iter(tx, *table_id)?.map(|_| 1u32).sum()))
+    }
+
+    fn empty_transaction(&mut self) -> ResultBench<()> {
+        self.db.with_auto_commit(|_tx| Ok(()))
+    }
+
+    type PreparedInsert<T> = PreparedQuery;
+    #[inline(never)]
+    fn prepare_insert<T: BenchTable>(&mut self, table_id: &Self::TableId) -> ResultBench<Self::PreparedInsert<T>> {
+        Ok(PreparedQuery { table_id: *table_id })
+    }
+
+    fn insert<T: BenchTable>(&mut self, prepared: &Self::PreparedInsert<T>, row: T) -> ResultBench<()> {
+        self.db.with_auto_commit(|tx| {
+            self.db.insert(tx, prepared.table_id, row.into_product_value())?;
+            Ok(())
+        })
+    }
+
+    type PreparedInsertBulk<T> = PreparedQuery;
+    #[inline(never)]
+    fn prepare_insert_bulk<T: BenchTable>(
+        &mut self,
+        table_id: &Self::TableId,
+    ) -> ResultBench<Self::PreparedInsertBulk<T>> {
+        Ok(PreparedQuery { table_id: *table_id })
+    }
+
+    fn insert_bulk<T: BenchTable>(&mut self, prepared: &Self::PreparedInsertBulk<T>, rows: Vec<T>) -> ResultBench<()> {
+        self.db.with_auto_commit(|tx| {
+            for row in rows {
+                self.db.insert(tx, prepared.table_id, row.into_product_value())?;
+            }
+            Ok(())
+        })
+    }
+
+    type PreparedInterate = PreparedQuery;
+    #[inline(never)]
+    fn prepare_iterate<T: BenchTable>(&mut self, table_id: &Self::TableId) -> ResultBench<Self::PreparedInterate> {
+        Ok(PreparedQuery { table_id: *table_id })
+    }
+    #[inline(never)]
+    fn iterate(&mut self, prepared: &Self::PreparedInterate) -> ResultBench<()> {
+        self.db.with_auto_commit(|tx| {
+            for row in self.db.iter(tx, prepared.table_id)? {
+                black_box(row);
+            }
+            Ok(())
+        })
+    }
+
+    type PreparedFilter = PreparedFilter;
+    #[inline(never)]
```

Comment:
I picked this `#[inline(never)]` one arbitrarily out of all of them to comment: you should describe where and why you do this in some top level place if possible!